App/views.py

```diff
@@ -0,0 +1,79 @@
+import json
+import requests
+import urllib.parse
+
+from django.http import HttpResponse
+from django.shortcuts import render
+
+
+def user_revisions_analysis(request):
+    """
+    Display analysis of last 5 edits made by user.
+    """
+    if request.method=='POST':
+        # Here, we get the username
+        username = request.POST['username']
+        # if username is submitted blank 
+        if not username:
+            return render(request, "App/task4.html", {"error": "Please enter a username"})
+
+        # if username is not blank
+        details = list()
+        parameters = {'action':'query',
+                      'format':'json',
+                      'list':'usercontribs',
+                      'uclimit':'5',
+                      'ucuser':username}
+        url_parameters = urllib.parse.urlencode(parameters)
+        url1 = 'https://en.wikipedia.org/w/api.php?{}'.format(url_parameters)
+        
+        # Recieved data in json-format
+        response = requests.get(url1)
+ 
+        if response.status_code == 200:
+            lst = json.loads(response.text)
+            try:
```

Comment:
This is a weird way to use exceptions. Normally you would put the normal behavior in the try branch and the error handling in the except branch. In part because it's easier for others to understand your thinking that way, in part because the main branch tends to be longer and involve multiple types of errors. So now you have an except branch which itself can throw exceptions... not a good thing.

Also, specifically for dictionary existence checks, you can just use `'error' in lst` or `lst.get('error')`, neither of which can throw exceptions.

Reply:
Yes sir, it should have been way. I have corrected it now.